documentation/src/main/asciidoc/userguide/appendices/LegacyBasicTypeResolution.adoc

```diff
@@ -0,0 +1,346 @@
+
+[[basic-type-annotation-example]]
+.Using `@org.hibernate.annotations.Type`
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/basic/ExplicitTypeTest.java[tags=basic-type-annotation-example]
+----
+====
+
+This tells Hibernate to store the Strings as nationalized data.
+This is just for illustration purposes; for better ways to indicate nationalized character data see <<basic-nationalized>> section.
+
+Additionally, the description is to be handled as a LOB. Again, for better ways to indicate LOBs see <<basic-lob>> section.
+
+The `org.hibernate.annotations.Type#type` attribute can name any of the following:
+
+* Fully qualified name of any `org.hibernate.type.Type` implementation
+* Any key registered with `BasicTypeRegistry`
+* The name of any known _type definitions_
+
+[[basic-custom-type]]
+=== Custom BasicTypes
+
+Hibernate makes it relatively easy for developers to create their own basic type mappings type.
+For example, you might want to persist properties of type `java.util.BigInteger` to `VARCHAR` columns, or support completely new types.
+
+There are two approaches to developing a custom type:
+
+- implementing a `BasicType` and registering it
+- implementing a `UserType` which doesn't require type registration
+
+As a means of illustrating the different approaches, let's consider a use case where we need to support a `java.util.BitSet` mapping that's stored as a VARCHAR.
+
+[[basic-custom-type-BasicType]]
+==== Implementing a `BasicType`
+
+The first approach is to directly implement the `BasicType` interface.
+
+[NOTE]
+====
+Because the `BasicType` interface has a lot of methods to implement, if the value is stored in a single database column, it's much more convenient to extend the `AbstractStandardBasicType` or the `AbstractSingleColumnStandardBasicType` Hibernate classes.
+====
+
+First, we need to extend the `AbstractSingleColumnStandardBasicType` like this:
+
+[[basic-custom-type-BitSetType-example]]
+.Custom `BasicType` implementation
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/basic/BitSetType.java[tags=basic-custom-type-BitSetType-example]
+----
+====
+
+The `AbstractSingleColumnStandardBasicType` requires an `jdbcTypeDescriptor` and a `javaTypeDescriptor`.
+The `jdbcTypeDescriptor` is `VarcharTypeDescriptor.INSTANCE` because the database column is a VARCHAR.
+On the Java side, we need to use a `BitSetJavaType` instance which can be implemented like this:
+
+[[basic-custom-type-BitSetTypeDescriptor-example]]
+.Custom `AbstractTypeDescriptor` implementation
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/basic/BitSetTypeDescriptor.java[tags=basic-custom-type-BitSetTypeDescriptor-example]
+----
+====
+
+The `unwrap` method is used when passing a `BitSet` as a `PreparedStatement` bind parameter, while the `wrap` method is used to transform the JDBC column value object (e.g. `String` in our case) to the actual mapping object type (e.g. `BitSet` in this example).
```

Comment:
I prefer to put parens in method names: `unwrap()` and `wrap()`. And I think that's pretty standard.